File: JavaNgArch/src/projectbase/utils/ProjectHierarchy.java
package projectbase.utils;

import java.util.Map;

import org.apache.commons.lang3.StringUtils;

import projectbase.sharparch.domain.designbycontract.Check;

public class ProjectHierarchy {
	private static String _businessDelegateNS;
	private static String _dataNS;
	private static String _domainNS;
	private static String _mvcNS;
	private static Map<String,String> _namespaceMapToTablePrefix;

	public static String ProjectName = null;

	public static String getBusinessDelegateNS() {

		if (_businessDelegateNS == null || _businessDelegateNS.isEmpty())
			return ProjectName + ".businessdelegate";
		else {
			return _businessDelegateNS;
		}
	}

	public static void setBusinessDelegateNS(String value) {
		_businessDelegateNS = value;
	}

	public static String getDataNS() {
		if (_dataNS == null || _dataNS.isEmpty())
			return ProjectName + ".data";
		else {
			return _dataNS;
		}
	}

	public static void setDataNS(String value) {
		_dataNS = value;
	}

	public static String getDomainNS() {
		if (_domainNS == null || _domainNS.isEmpty())
			return ProjectName + ".domain";
		else {
			return _domainNS;
		}
	}

	public static void setDomainNS(String value) {
		_domainNS = value;
	}

	public static String getMvcNS() {

		if (_mvcNS == null || _mvcNS.isEmpty())
			return ProjectName + ".mvc";
		else {
			return _mvcNS;
		}
	}
	public static void setMvcNS(String value) {
		_mvcNS = value;
	}

	public static Map<String,String> getNamespaceMapToTablePrefix() {
		return _namespaceMapToTablePrefix;
	}
	public static void setNamespaceMapToTablePrefix(Map<String,String> value)
    {
		_namespaceMapToTablePrefix=value;
    }
	public static void Init(String projectname,Map<String,String> namespaceMapToTablePrefix){
		Check.Require(StringUtils.isNotBlank(projectname));
		Check.Require(namespaceMapToTablePrefix!=null);
		ProjectName=projectname;
		_namespaceMapToTablePrefix=namespaceMapToTablePrefix;
	}
	public static String MessagesResourceClassKey = "Messages";
	public static String ValidationMessagesResourceClassKey = "ValidationMessages";
	public static String DisplayNameResourceClassKey = "DisplayName";
}

File: NetNgArch/src/ProjectBase/Utils/ProjectHierarchy.cs

namespace ProjectBase.Utils
{
    public class ProjectHierarchy
    {
        private static string _businessDelegateNS;
        private static string _dataNS;
        private static string _domainNS;
        private static string _mvcNS;

        public static string ProjectName = "App";

        public static string BusinessDelegateNS
        {
            get
            {
                if (string.IsNullOrEmpty(_businessDelegateNS))
                    return ProjectName + ".BusinessDelegate";
                else
                {
                    return _businessDelegateNS;
                }
            }
            set { _businessDelegateNS = value; }

        }
        public static string DataNS
        {
            get
            {
                if (string.IsNullOrEmpty(_dataNS))
                    return ProjectName + ".Data";
                else
                {
                    return _dataNS;
                }
            }
            set { _dataNS = value; }

        }
        public static string DomainNS
        {
            get
            {
                if (string.IsNullOrEmpty(_domainNS))
                    return ProjectName + ".Domain";
                else
                {
                    return _domainNS;
                }
            }
            set { _domainNS = value; }

        }
        public static string MvcNS
        {
            get
            {
                if (string.IsNullOrEmpty(_mvcNS))
                    return ProjectName + ".Web.Mvc";
                else
                {
                    return _mvcNS;
                }
            }
            set { _mvcNS = value; }

        }

        public static string MessagesResourceClassKey = "Messages";
        public static string ValidationMessagesResourceClassKey = "ValidationMessages";
        public static string DisplayNameResourceClassKey = "DisplayName";
    }
}
